manager/dispatcher/dispatcher.go

```diff
@@ -328,14 +328,21 @@ func (d *Dispatcher) isRunning() bool {
 }
 
 // register is used for registration of node with particular dispatcher.
-func (d *Dispatcher) register(ctx context.Context, nodeID string, description *api.NodeDescription) (string, error) {
+func (d *Dispatcher) register(ctx context.Context, nodeID string, request *api.SessionRequest) (string, error) {
+	// break out the description for easier use later
+	description := request.Description
+
 	// prevent register until we're ready to accept it
 	if err := d.isRunningLocked(); err != nil {
 		return "", err
 	}
 
-	if err := d.nodes.CheckRateLimit(nodeID); err != nil {
-		return "", err
+	// check if we're trying to re-register an old session
+	// so, if the sessionid is not nil and is known to the dispatcher
+	if _, err := d.nodes.GetWithSession(nodeID, request.SessionID); err != nil {
```

Comment:
Bah! Sorry.

It is sufficient to use the return value from here to resume the existing session.

This might be easier to control from the `Session()` method.